### PR TITLE
draft of migration

### PR DESCRIPTION
**Pull request by ivanovSPvirtru**

### CODE REVIEW

File: lib/package.json [hunk 1]
```diff
@@ -105,15 +106,15 @@
     "@web/test-runner-commands": "~0.6.5",
     "audit-ci": "^6.3.0",
     "babel-loader": "^8.2.5",
-    "babel-plugin-module-resolver": "^4.1.0",
```
Comment (ivanovSPvirtru) [inline lib/package.json:108]: ```fix available via `npm audit fix --force`
Will install babel-plugin-module-resolver@2.2.0, which is a breaking change```
Thats for audit issue

File: lib/src/auth/Eas.ts [hunk 2]
```diff
@@ -35,7 +34,7 @@ class Eas {
     endpoint: string;
     requestFunctor?: RequestFunctor;
   }) {
-    this.authProvider = authProvider.withCreds;
+    this.authProvider = authProvider;
```
Comment (ivanovSPvirtru) [inline lib/src/auth/Eas.ts:37]: There were bug with context (this) cause it was executed from other object.

File: lib/src/auth/auth.ts [hunk 3]
```diff
@@ -14,16 +14,24 @@ export type Method =
 /**
  * Generic HTTP request interface used by AuthProvider implementers.
  */
-export type HttpRequest = {
+export class HttpRequest {
```
Comment (ivanovSPvirtru) [inline lib/src/auth/auth.ts:17]: We use it like this so I had to make it class back

File: lib/tdf3/src/tdf.ts [hunk 4]
```diff
@@ -300,7 +300,7 @@ class TDF extends EventEmitter {
       }
       encryptionInformation.keyAccess.push(keyAccess);
     }
-
+    debugger;
```
Comment (dmihalcik-virtru) [inline lib/tdf3/src/tdf.ts:303]:
```suggestion
```

**Review by dmihalcik-virtru — APPROVED** [review]

**Review by dmihalcik-virtru — APPROVED** [review]

**Pull request merged**